### PR TITLE
Split out code which creates skopeo command into helper function

### PR DESCRIPTION
**Pull request by cgwalters**
So it can be unit tested.

### CODE REVIEW

File: src/imageproxy.rs [hunk 1]
```diff
@@ -149,16 +150,10 @@ pub struct ImageProxyConfig {
     pub skopeo_cmd: Option<Command>,
 }
 
-impl ImageProxy {
-    /// Create an image proxy that fetches the target image, using default configuration.
-    pub async fn new() -> Result<Self> {
-        Self::new_with_config(Default::default()).await
-    }
+impl TryFrom<ImageProxyConfig> for Command {
+    type Error = anyhow::Error;
```
Comment (jlebon) [inline src/imageproxy.rs:154]: Shouldn't libraries use `thiserror` instead?

File: src/imageproxy.rs [hunk 2]
```diff
@@ -378,3 +388,45 @@ impl ImageProxy {
...
+mod tests {
+    use super::*;
+
+    fn validate(c: Command, contains: &[&str], not_contains: &[&str]) {
+        // Format via debug, because
+        // https://doc.rust-lang.org/std/process/struct.Command.html#method.get_args
+        // is experimental
+        let d = format!("{:?}", c);
+        for c in contains {
+            assert!(d.contains(c), "missing {}", c);
```
Comment (jlebon) [inline src/imageproxy.rs:402]: Seems like the panic msg should include `d`.

File: src/imageproxy.rs [hunk 3]
```diff
@@ -378,3 +388,45 @@ impl ImageProxy {
...
+        }
+        for c in not_contains {
+            assert!(!d.contains(c), "should not contain {}", c);
```
Comment (jlebon) [inline src/imageproxy.rs:405]: And here too.

**Review by jlebon — APPROVED** [review]
Minor comments, LGTM overall.

File: src/imageproxy.rs (hunk 1 above)
Comment (cgwalters) [reply src/imageproxy.rs:154]: Mmm, I need to write a blog on that.

Comment (cgwalters) [reply src/imageproxy.rs:154]: https://users.rust-lang.org/t/why-i-use-anyhow-error-even-in-libraries/68592

**Pull request merged**